src/python/pants/backend/python/lint/black/subsystem.py

```diff
@@ -107,9 +113,23 @@ class BlackLockfileSentinel(PythonToolLockfileSentinel):
     pass
 
 
-@rule
-def setup_black_lockfile(_: BlackLockfileSentinel, black: Black) -> PythonLockfileRequest:
-    return PythonLockfileRequest.from_tool(black)
+@rule(
+    desc="Determine if Black should use Python 3.8+",
+    level=LogLevel.DEBUG,
+)
+async def setup_black_lockfile(
+    _: BlackLockfileSentinel, black: Black, python_setup: PythonSetup
+) -> PythonLockfileRequest:
```

Comment:
Nit: the disconnect between the description and the rule purpose gets to the fact that when you "set up a lockfile", what you're actually doing is "setting up a resolve", and the lockfile is the on-disk representation of the resolve. When someone wants to consume the resolve, they'll need to make a request with a very similar shape.

Moving forward, I expect the terminology will move in the direction of "requesting a resolve" (with an optional lockfile location field), but it's possible it should affect the API sooner.

Reply:
Yeah I think that Chris's cache invalidation work is going to take us in that direction: consumers of lockfiles will need to know what the LockfileRequest is so that they can see if the hash is different from what is persisted to disk (via a header comment).